### ISSUE TITLE
[Bug] SymbolLayer crashes on style update

### ISSUE THREAD

**Issue by wafisher** [visgl/react-map-gl#1844]
### Description

When using `Source/Layer` with a `SymbolLayer` and a (modified) Immutable base style, I get the following in the console.

```
mapbox-gl.js:35 Uncaught TypeError: Cannot read properties of undefined (reading 'get')
    at new It (mapbox-gl.js:35:1)
    at zt.continuePlacement (mapbox-gl.js:35:1)
    at Nt._updatePlacement (mapbox-gl.js:35:1)
    at Map._render (mapbox-gl.js:35:1)
    at Map.map._render (mapbox.ts:552:1)
    at mapbox-gl.js:35:1
```

There's some interaction with both of these that brings about the issue. If I remove the SymbolLayer, I don't get this issue. And if I don't modify the style, I also don't get the issue.

### Steps to Reproduce

I've created a minimal reproduction here: https://codesandbox.io/s/great-mccarthy-0pozvg. Click on the map to trigger the issue.

### Environment

- Framework version: 7.0.11
- Browser: latest Chrome and Safari

**Comment by Pessimistress** [comment]
I can reproduce the crash with native mapbox-gl: https://codesandbox.io/s/proud-wave-gohost

You may wish to report this to the [Mapbox GL JS repo](https://github.com/mapbox/mapbox-gl-js/issues).

**Comment by wafisher** [comment]
Done https://github.com/mapbox/mapbox-gl-js/issues/11820

**Comment by wafisher** [comment]
I believe this is fixed with the new release?

Just confirming: #2047 is strictly a workaround because https://github.com/mapbox/mapbox-gl-js/issues/11820 has not been resolved, correct?

**Comment by Pessimistress** [comment]
You are correct that the commit is only a workaround. We can revisit if Mapbox fixes the issue.

Closing as resolved as of v7.0.20.
